lib/pacemaker/pcmk_sched_location.c

```diff
@@ -253,7 +253,8 @@ generate_location_rule(pcmk_resource_t *rsc, xmlNode *rule_xml,
 static void
 unpack_rsc_location(xmlNode *xml_obj, pcmk_resource_t *rsc,
                     const char *role_spec, const char *score,
-                    pe_re_match_data_t *re_match_data)
+                    char *rsc_id_match, int rsc_id_nmatches,
```

Comment:
A `pe_re_match_data_t` struct really does suit this well... these three values (almost?) always appear together. I don't think they ever make sense without having all three of them.

I think I understand why you're doing this though. In `pcmk_rule_input_t`, we've already flattened it out from a nested struct member to three individual members. I think that was mostly for two reasons:
* to avoid having to NULL-check before dereferencing
* so that we don't have to maintain another public struct definition (for `pe_re_match_data_t`).

But I'm starting to question whether that was a good idea.

It's fine either way, I don't want to get hung up on it.